SCons/Tool/msvc.py

```diff
@@ -99,6 +99,9 @@ def pch_emitter(target, source, env):
 
     target = [pch, obj] # pch must be first, and obj second for the PCHCOM to work
 
+    if 'PCH' not in env or not env['PCH']:
+        env['PCH'] = pch
+
```

Comment:
If this really needs both parts of the check I guess it's fine. The common idiom when you want to preserve any existing value (but *not* to also test what it is):
```py
    env.SetDefault(PCH=pch)
```

Reply:
I haven't looked at this in quite awhile.

There was a comment in the issue #2249 (https://github.com/SCons/scons/issues/2249#issuecomment-1836436706) where there appeared to be a reason evaluating False was important:

> > I wonder if the idea is to say not to set the PCH consvar manually, and instead have the PCH builder be the one that sets it.
>
> The suggested solution was adjusted to set `env['PCH']=pch` in the msvc pch_emitter when either `env['PCH']` is undefined or evaluates to false.

I'll have to dig back in one of these days.  Unfortunately, available time is limited in the immediate short term.

Work was suspended on this for awhile because I was convinced there was a situation/confluence of events that needed state to be saved and/or maintained which is problematic.  Somewhere here I had some prototype code that looked promising.